stac_validator/validate.py

```diff
@@ -267,75 +271,100 @@ def recursive_validator(self, stac_type: str) -> bool:
                         click.echo(json.dumps(message, indent=4))
         return True
 
-    def validate_dict(cls, stac_content):
-        cls.stac_content = stac_content
-        return cls.run()
+    def validate_dict(self, stac_content):
+        self.stac_content = stac_content
+        return self.run()
 
-    def run(cls):
+    def validate_item_collection_dict(self, item_collection):
+        for item in item_collection["features"]:
+            self.custom = ""
+            self.validate_dict(item)
+
+    def validate_item_collection(self):
+        print("processing page 1")
+        item_collection = fetch_and_parse_file(self.stac_file)
+        self.validate_item_collection_dict(item_collection)
+        try:
+            if self.pages is not None:
+                for x in range(self.pages - 1):
+                    if "links" in item_collection:
+                        for link in item_collection["links"]:
+                            if link["rel"] == "next":
+                                next_link = link["href"]
+                                self.stac_file = next_link
+                                print(f"processing page {x+2}")
+                                item_collection = fetch_and_parse_file(self.stac_file)
+                                self.validate_item_collection_dict(item_collection)
+                                break
+        except Exception as e:
+            error_msg = ("Error is in pagination: ", str(e))
+            return error_msg
```

Comment:
Is there a reason why a string is returned, rather than allowing the error to propagate up? Is this error string reported out to the user somehow?

Reply:
No. You're right. I just got confused about what would be happening with an error following the pagination links.